ApollosProject/apollos-prototype
# Adds Terraform Config

## DESCRIPTION

### What does this PR do, or why is it needed?

This PR adds a terraform setup to create Heroku infrastructure using Terraform. By copying and pasting the `api_heroku.tf` file,  any of our clients could quickly provision a Heroku setup with,

- Fastly installed
- Cloudinary installed and setup
- Environmental Variable Placeholders setup in Heroku dashboard
- `container` proctype configured

### What design trade-offs/decisions were made?

This PR **does not** setup any sort of deployment or building via terraform. This is something I explored with Marcello, and decided that it is out of the scope of what Terraform should be doing. It was technically doable with Terraform, but in the process of exploring how to improve the setup I discovered that it's not recommended that Terraform is involved in the actual deployment of application code. 

### How do I test this PR?

1. Pull this PR.
2. Follow the steps in "**How**" in the `TERRAFORM.md`
3. At the end you should have a project setup in Heroku, ready to deploy code to. You can remove this by running `terraform destroy`

### What automated tests did you write?

n/a

## TODO:

- [x] Closes #487 

## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
